cmd/signozcollector/Dockerfile

```diff
@@ -1,34 +1,27 @@
-# Builder stage
-FROM golang:1.18-alpine as build
-RUN apk --update add ca-certificates
+# use a minimal alpine image
+FROM alpine:3.17
 
-WORKDIR /src
-COPY . .
+# add ca-certificates in case you need them
+RUN apk update && apk add ca-certificates && rm -rf /var/cache/apk/*
 
-ARG TARGETPLATFORM
+# define arguments and default values
+ARG TARGETOS=linux
+ARG TARGETARCH=amd64
 
-RUN export GOOS=$(echo ${TARGETPLATFORM} | cut -d / -f1) && \
-    export GOARCH=$(echo ${TARGETPLATFORM} | cut -d / -f2)
+# copy the binaries from the multi-stage build
+COPY .build/${TARGETOS}-${TARGETARCH}/signoz-collector /signoz-collector
```

Comment:
Don't you have to set `TARGETOS` and `TARGETARCH` while building docker images?

Reply:
We do not have to set them. It is automatically set globally by buildkit.

Reference: https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope